spam-n-eggs/ansible-aws-transfer
# Updated to process SSH Keys

Details
===
This pull request adds the ability to manage SSH keys to the SFTP server.

Steps to test changes
===
1. Setup an [args.json](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general.html#exercising-module-code-locally) with the following arguments: 
**ssh_key**: *A Public key such as the one stored in ~/.ssh/id_rsa.pub*
**server_id**: *The Assigned ID of a valid SFTP Server in AWS*
**user_name**: *The user name of a pre-existing user on the SFTP server in AWS*
**state**:*Whether to add or remove the SSH key* Valid values are `present` and `absent`
**aws_access_key**:*An AWS Access Key for your IAM user or an IAM user that has permissions to modify a transfer server*
**aws_secret_key*:**The AWS Secret Key to go with the access key.*
1. Run the following line of code `python3 ansible_collections/spam_n_eggs/amazon/plugins/modules/transfer_user_credentials.py args.json`
1. Manually verify that the ssh key has been created on the specified AWS Transfer User.
1. Set state to `absent` in args.json
1. Re-run the python3 call from above.
1. Manually verify that the ssh key has been removed from the AWS Transfer User
